### ISSUE TITLE
build errors uhd 4.0rc1 and boost 1_67_0

### ISSUE THREAD

**Issue by rcarmich** [EttusResearch/uhd#369]
## Issue Description ##
Building UHD 4rc1 with boost 1_67_0 results in

```
boost_1_67_0/include/boost/lockfree/spsc_queue.hpp:140:43: error: ‘next’ is not a member of ‘boost’
```
## Expected Behavior ##
Compilation succeeds

## Actual Behaviour ##
Compilation fails
```
boost_1_67_0/include/boost/lockfree/spsc_queue.hpp:140:43: error: ‘next’ is not a member of ‘boost’
```
## Fix ##
spsc_queue.hpp should include:

```
#include <boost/fusion/iterator/next.hpp>
#include <boost/fusion/include/next.hpp>
```

and fully reference

``boost::fusion::next``

**Comment by rcarmich** [comment]
Let me revise this with a likely better suggestion - if you just include <boost/next_prior.hpp> then boost::next works without going through ::fusion.

**Comment by rcarmich** [comment]
Let me revise this with a likely better suggestion - if you just include <boost/next_prior.hpp> then boost::next works without going through ::fusion.

**Comment by mbr0wn** [comment]
@rcarmich Thansk for the feedback. FYI, we'll need a solution that works all the way down to Boost 1.58, so we need to make sure we're not breaking older Boosts which is usually the hold-up for things like this.

**Comment by mbr0wn** [comment]
@rcarmich So a couple of odd things. First, you're suggesting to change a file in Boost itself -- was that intentional? Second, `boost::next` is never used in UHD 4.0. Third, we've tested UHD with a bunch of Boost versions, not sure if 1.67 was one of them, but newer versions are. Can you provide more info what you were testing? For example, where did that Boost come from. Is it part of a regular Ubuntu/Fedora/CentOS/Whatever system?

**Comment by rcarmich** [comment]
@mbr0wn  you're right, that was my mistake - I shouldn't have suggested that fix, please ignore. But I believe the issue definitely exists.

Here's an easy way to reproduce (excuse the inefficient Dockerfile) - see attached.
[Dockerfile.txt](https://github.com/EttusResearch/uhd/files/5197217/Dockerfile.txt)

**Comment by mbr0wn** [comment]
Fix is on the way.

**Comment by lv254165754** [comment]
You should see this:
https://github.com/boostorg/lockfree/commit/12726cda009a855073b9bedbdce57b6ce7763da2?diff=split
